### ISSUE TITLE
JSpecify experimental: Override checks read the raw JDK signature, not the model

### ISSUE THREAD

**Issue by msridhar** [uber/NullAway#1718]
(1 from https://github.com/ben-manes/caffeine/issues/2004#issuecomment-5382301640)

At a **call site** the astubx annotations are in play. At an **override site** the same method's
supertype is reported unannotated. Both behaviours in one class:

```java
abstract class T implements Map<String, String> {
  static void callSite(Map<String, String> m,
      BiFunction<String, @Nullable String, @Nullable String> good,
      BiFunction<String, String, String> bad) {
    m.compute("k", good);   // accepted
    m.compute("k", bad);    // error: ... cannot be converted to
                            //   BiFunction<? super String, ? super @Nullable String,
                            //              ? extends @Nullable String>
  }

  // The signature the call site just demanded:
  @Override
  public @Nullable String compute(String key,
      BiFunction<? super String, ? super @Nullable String, ? extends @Nullable String> f) {
    return null;
  }
  // error: method returns @Nullable, but superclass method
  //   java.util.Map.compute(K,BiFunction<? super K,? super V,? extends V>) returns @NonNull
  // error: Parameter has type BiFunction<? super String, ? super @Nullable String,
  //   ? extends @Nullable String>, but overridden method has parameter type
  //   BiFunction<? super String, ? super String, ? extends String>

  @Override public abstract @Nullable String get(@Nullable Object key);   // accepted
}
```

jspecify/jdk declares `@Nullable V compute(K, BiFunction<? super K, ? super @Nullable V,
? extends @Nullable V>)`, so the call site is right and the override site is not. `get` is
accepted only because `Map.get` is also in NullAway's hardcoded `DefaultLibraryModels`, which the
override path does consult — that is the tell for where the two paths diverge.
